src/targets/browser/serviceWorkerRegistration.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-console */
+/* eslint-disable promise/always-return */
+/* eslint-disable promise/catch-or-return */
+const isLocalhost = Boolean(
+  window.location.hostname.endsWith('cozy.localhost') ||
+    window.location.hostname.endsWith('cozy.tools')
+)
+
+export function register() {
+  if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
+    window.addEventListener('load', () => {
+      const swUrl = `/service-worker.js`
+
+      if (isLocalhost) {
+        // This is running on localhost. Let's check if a service worker still exists or not.
+        checkLocalhostValidServiceWorker(swUrl)
+
+        // Add some additional logging to localhost, pointing developers to the
+        // service worker/PWA documentation.
+        navigator.serviceWorker.ready.then(() => {
+          console.log(
+            'This web app is being served cache-first by a service ' +
+              'worker. To learn more, visit https://cra.link/PWA'
+          )
+        })
+      } else {
+        // Is not localhost. Just register service worker
+        registerValidSW(swUrl)
+      }
+    })
+  }
```

Comment:
So if the node_env is dev then we don't load the service-worker?

Reply:
exactly. No service-worker, when yarn watch or yarn test

Reply:
Can you explain why you do that? Hot reload issue? 

My first impression is that publishing something we don't use while developing can be error prone, no?

Reply:
https://github.com/facebook/create-react-app/issues/2396#issuecomment-304539651

Reply:
First of all, I see a lot of projects that don't use service worker in local development. It's not their purpose to be used there.

Moreover, that would imply to run the inject manifest steps at every watch (it will slow down dev).

Reply:
Apparently, in the CRA, https://stackoverflow.com/questions/65063966/how-to-use-service-worker-in-dev-mode , that would mean to change the workbox config   
```
dontCacheBustURLsMatching: /\.[0-9a-f]{8}\./, 
```
https://developer.chrome.com/docs/workbox/reference/workbox-build/#type-WebpackInjectManifestOptions => look for `dontCacheBustURLsMatching`

Reply:
in summary, there will be difference between dev and production mode, and I believe it's simpler not to handle service worker in development.

Reply:
So there is a new step now before submitting a PR: 
- Build your app in production mode and check if it works correctly with the SW. Right?

Reply:
I don't see why this would break anything (except if we play with service-worker config)